jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebAppClassLoader.java

```diff
@@ -628,12 +615,47 @@ public boolean removeTransformer(ClassFileTransformer transformer)
         return _transformers.remove(transformer);
     }
 
+    
+    /**
+     * Look for the classname as a resource to avoid loading a class that is 
+     * potentially a system resource.
+     * 
+     * @param name the name of the class to load
+     * @param checkSystemResource if true and the class isn't a system class we return it
+     * @return the loaded class
+     * @throws ClassNotFoundException
+     */
+    protected Class<?> loadAsResource (final String name, boolean checkSystemResource) throws ClassNotFoundException
+    {
+        // Try the webapp classloader first
+        // Look in the webapp classloader as a resource, to avoid 
+        // loading a system class.
+        Class<?> webapp_class = null;
+        String path = name.replace('.', '/').concat(".class");
+        URL webapp_url = findResource(path);
+        
+        if (webapp_url!=null && (!checkSystemResource || !_context.isSystemResource(name,webapp_url)))
+        {
+
+            webapp_class = this.foundClass(name,webapp_url);
+            resolveClass(webapp_class);
```

Comment:
I don't think we should to the resolveClass here, it may be that the class is being loaded as a resource to look at it and not load it!

Reply:
The loadAsResource() method is a protected method of WebAppClassLoader. It is only ever called by the loadClass() method, the whole point of which is to actually load the class, so I don't think this is a concern?